### PR TITLE
feat: enhance user query using LLM

### LINKED ISSUE #4: [feat] Enhance User Prompt by using any LLM as NLP
## Details:

The generation output is dependent totally upon the prompt, thus an LLM can be used as an intermediary which can further enhance the prompt. This will result in a better output GIF, this will be an optional feature and can be easily disabled from the settings panel.

### PR DESCRIPTION
**Pull request by AmanNegi**
## Description

- Implement LLM as NLP to improve the user query. The LLM provides with 4 different variations of the query. The user can click on any of the variation to use as the prompt for GIF generation.
- Also provides a mock function for testing LLM functionality on LLM-inacessible servers.

> Fixes: #4 

## Preview

https://github.com/RocketChat/Apps.AI.GIF/assets/37607224/796b05ee-6c99-4dec-a848-a365993d53a7

### CODE REVIEW

File: src/commands/GenGifCommand.ts [hunk 1]
```diff
@@ -93,4 +94,86 @@ export class GenGifCommand implements ISlashCommand {
...
+                http,
+                context.getSender().id,
+                this.app.getLogger()
+            );
+
+            return data;
+        },
+        2000
+    );
+
```
Comment (Nabhag8848) [inline src/commands/GenGifCommand.ts]: Can you make a seperate class / service for debouncing + make this typesafe and add comments more comments.

**Review by Nabhag8848 — CHANGES_REQUESTED** [review]
@AmanNegi added comments, Also can you share demo again here around.
1. What happens in this case when the LLM doesn't response.
2. What happens when we don't get the response from the model Itself.

**Comment by AmanNegi** [comment]
> @AmanNegi added comments, Also can you share demo again here around.
> 
>     1. What happens in this case when the LLM doesn't response.
> 
>     2. What happens when we don't get the response from the model Itself.

In these cases where the LLM/Model doesn't response for too long, the HTTP request will expire and it will throw an error. For the end user he would not know about the error, and he wouldn't be getting anything on the preview. 

Ps: Currently I am not handling the reject promise case, I'll do it in the next commit.

> The error can be identified through logs, and alternatively we could send the error message as notification to the user as well. Let me know if we expect that behavior.

**Comment by AmanNegi** [comment]
@Nabhag8848 I have made the request changes. In the implementation in case the sever doesn't respond or there is any other kind of error from server side, the user will be notified via a message(visible to self).

#### Demo

https://github.com/RocketChat/Apps.AI.GIF/assets/37607224/8b629f3b-eb11-4236-a163-51a45d730479

**Review by Nabhag8848 — APPROVED** [review]
LGTM 🚀

**Pull request merged**